resourcemanager/commonschema/location.go

```diff
@@ -3,8 +3,20 @@ package commonschema
 import (
 	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 )
 
+func ExtendedLocation() *schema.Schema {
+	return &schema.Schema{
+		Type:             schema.TypeString,
+		Required:         true,
+		ForceNew:         true,
+		StateFunc:        location.StateFunc,
+		DiffSuppressFunc: location.DiffSuppressFunc,
+		ValidateFunc:     validation.StringIsNotEmpty,
```

Comment:
As there is a [list API](https://docs.microsoft.com/en-us/rest/api/resources/subscriptions/list-locations) to retrieve available extended locations, but it has to be whitelisted by service team first before calling this API. So maybe here can add a property validation for extendedLocation like the [location validation](https://github.com/hashicorp/go-azure-helpers/blob/41ef58a3f0054aba44bb4231c2063682395fca58/resourcemanager/commonschema/location.go#L13) when this property is treated as optional property otherwise it would break existing users/resources. Does it make sense?